lib/iron_bank/authentications/token.rb

```diff
@@ -58,9 +58,9 @@ def authenticate
 
       def connection
```

Comment:
Do we know why we create our own connection here, instead of leveraging `IronBank::Client`? cc: @mickaelpham, @mustafaturan

Reply:
Yeah, the authentication for the `OAuth` token needs:

1. request to be `url_encoded`
2. does not use the `bearer` token